lib/membrane/video_compositor/video_compositor.ex

```diff
@@ -4,99 +4,95 @@ defmodule Membrane.VideoCompositor do
-    accepted_format: %Membrane.RawVideo{pixel_format: :I420},
+    accepted_format: %RawVideo{pixel_format: :I420},
     availability: :on_request,
     options: [
-      initial_placement: [
-        spec: BaseVideoPlacement.t(),
-        description: "Initial placement of the video on the screen"
+      video_config: [
+        spec: VideoConfig.t(),
+        description: "Specify how single input video should be transformed"
       ],
       timestamp_offset: [
         spec: Membrane.Time.non_neg_t(),
         description: "Input stream PTS offset in nanoseconds. Must be non-negative.",
         default: 0
-      ],
-      initial_video_transformations: [
-        spec: VideoTransformations.t(),
-        description:
-          "Specify the initial types and the order of transformations applied to video.",
-        # Can't set here struct, due to quote error (AST invalid node).
-        # Calling Macro.escape() returns tuple and makes code more error prone and less readable.
-        default: nil
       ]
     ]
 
   def_output_pad :output,
-    accepted_format: %Membrane.RawVideo{pixel_format: :I420},
+    accepted_format: %RawVideo{pixel_format: :I420},
     availability: :always
 
   @impl true
-  def handle_init(_ctx, options) do
+  def handle_init(
+        _ctx,
+        options = %{output_stream_format: output_stream_format = %RawVideo{framerate: framerate}}
+      ) do
     spec =
-      child(:compositor, %CompositorElement{
-        stream_format: options.stream_format
+      child(:queue, get_queue(options))
+      |> child(:compositor_core, %VC_Core{
+        output_stream_format: output_stream_format
       })
       |> bin_output()
 
-    state = %{output_stream_format: options.stream_format}
+    state = %{output_framerate: framerate}
 
     {[spec: spec], state}
   end
 
-  @impl true
-  def handle_pad_removed(Pad.ref(:input, pad_id), _context, state),
-    do: {[remove_child: {:framerate_converter, pad_id}], state}
-
   @impl true
   def handle_pad_added(Pad.ref(:input, pad_id), context, state) do
     spec =
       bin_input(Pad.ref(:input, pad_id))
-      |> child({:framerate_converter, pad_id}, %FramerateConverter{
-        framerate: state.output_stream_format.framerate
-      })
       |> via_in(Pad.ref(:input, pad_id),
         options: [
-          initial_placement: context.options.initial_placement,
           timestamp_offset: context.options.timestamp_offset,
-          initial_video_transformations: context.options.initial_video_transformations
+          video_config: context.options.video_config
         ]
       )
-      |> get_child(:compositor)
+      |> get_child(:queue)
 
     {[spec: spec], state}
   end
 
   @impl true
-  def handle_parent_notification({:update_placement, placements}, _ctx, state) do
-    {[notify_child: {:compositor, {:update_placement, placements}}], state}
-  end
+  def handle_pad_removed(Pad.ref(:input, pad_id), _context, state),
+    do: {[remove_child: {:queue, pad_id}], state}
```

Comment:
this doesn't work...

Reply:
because `{:queue, pad_id}` is not a child of this bin. in the previous compositor, framerate converters were direct children and were called `{:framerate_converter, pad_id}`, that is why it worked then.